public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -260,10 +262,15 @@ export class ElasticDatasource
     );
   }
 
-  private prepareAnnotationRequest(options: { annotation: ElasticsearchAnnotationQuery; range: TimeRange }) {
+  private prepareAnnotationRequest(options: { annotation: ElasticsearchAnnotationQuery; dashboard: DashboardModel, range: TimeRange }) {
     const annotation = options.annotation;
     const timeField = annotation.timeField || '@timestamp';
     const timeEndField = annotation.timeEndField || null;
+    const dashboard = options.dashboard;
+
+    const adhocVariables = dashboard.getVariables().filter(v => v.type === "adhoc") as AdHocVariableModel[];
```

Comment:
```suggestion
    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
    const adhocVariables = dashboard.getVariables().filter(v => v.type === "adhoc") as AdHocVariableModel[];
```

Reply:
We have a lint rule to not be able to use assertions. In this case, it is needed and correct way to asset `AdHocVariableModel[]` so you'll need to add `// eslint-disable-next-line @typescript-eslint/consistent-type-assertions`